policy-controller/core/src/inbound.rs

```diff
@@ -26,21 +26,22 @@ pub enum AuthorizationRef {
 }
 
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
-pub enum HttpRouteRef {
+pub enum InboundRouteRef {
     Default(&'static str),
     Linkerd(GroupKindName),
 }
 
 /// Describes how a proxy should handle inbound connections.
-#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+#[derive(Clone, Debug, PartialEq, Eq)]
 pub enum ProxyProtocol {
     /// Indicates that the protocol should be discovered dynamically.
     Detect {
         timeout: Duration,
+        routes: HashMap<InboundRouteRef, InboundRoute<HttpRouteMatch>>,
```

Comment:
I don't think moving the routes inside the ProxyProtocol variants in the index is what we want, even though this is how we're eventually render them in the proxy API.  The reason is that we'll need to keep all route data for all protocols in the index in case the proxy protocol changes over time.  Consider the case where there exist HttpRoute resources AND GrpcRoute resources for a Server with proxyProtocol=http.  If we stored only the httproutes in the index, then if the proxyProtocol were changed to grpc, we wouldn't have the grpcroutes available.

Instead, I think it's better to keep the http_routes map on the InboundServer. When we add grpc support, we can add a grpc_routes field to InboundServer also.  Then, when rendering the proxy API message, we can choose which of these maps to use depending on the value of the proxy protocol.